components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -1414,4 +1418,50 @@ private org.wso2.carbon.user.core.common.User getUserFromUserID(String userId) {
                     ErrorMessage.ERROR_RETRIEVING_USERSTORE_MANAGER.getDescription());
         }
     }
+
+    /**
+     * This method will retrieve the user registrants configured for the given application id by calling the backend
+     * services.
+     *
+     * @param applicationId The application id
+     * @return the user registrants list object
+     */
+    public UserRegistrantsList getConfiguredUserRegistrants(String applicationId) {
+
+        String tenantDomain = ContextLoader.getTenantDomainFromContext();
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Retrieving the user registrants configured for the application," + applicationId);
+            }
+            List<AuthAttributeHolder> availableAuthAttributeHolders =
+                    ApplicationManagementServiceHolder.getAuthAttributeHandlerManager()
+                    .getAvailableAuthAttributeHolders(applicationId, tenantDomain);
+            List<UserRegistrant> userRegistrants = availableAuthAttributeHolders.stream().map(new
```

Comment:
can `availableAuthAttributeHolders` be null ?

Reply:
if there are no `AuthAttributeHolder`'s defined, the result will be an empty array